example/xml/consumer/tests/Service/HttpClientServiceTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace XmlConsumer\Tests\Service;
+
+use Tienvx\PactPhpXml\Model\Options;
+use Tienvx\PactPhpXml\XmlBuilder;
```

Comment:
We will probably want to document that this comes from a seperate-repo that would need to be added, into a readme of the changes.

https://github.com/tienvx/pact-php-xml

ps, what is the advantage of publishing these are separate packages? 

Is it useful for 

- users of the library?
- testing of the library?

or are there other advantages, outside of the possibly maintainence sprawl of having lots of different projects.

It might be that it works for PHP and is what users expect, but equally leads to more places to maintain, and more things to convey to the user, that they need, conditionally, depending on their particular use case

Reply:
The reason I put some code outside of this PR, into separate project `pact-php-xml` because:
* I just want to make this PR small so it easier to review
* The code of that library is highly opinionated
* That library is optional, can easily replace by raw json

What do you think if I move the code back to this PR?

Reply:
Not sure on a definitive answer, but good question and counter points. I've tagged a few other parties who may be offer to offer their thoughts.

On the fact its optional and highly opinionated, would you see this kind of library either coming under the pact-foundation, or us having a list of them in the repo, so that others can create their own and point to them (many ways to skin a cat and all that!) 

Hey team, just wondering if you can cast your eyes on the above thread. 🙏🏾 

@Greg0 @WaylandAce @Lewiscowles1986 @mmoll @cfmack

Reply:
reading now, but feel it would be better for Pact to have this repo under their umbrella with a contributor to review (there might be a lot I'd miss as I dip in and out)

Reply:
Regarding the specifics, it's a value-object using builder; it's a strange mix. Why this vs templating? Unless I'm mistaken (in which case I apologise); but it looks like a special string builder. XML is not strictly speaking a string.

Reply:
I believe it to be a special string builder, as we need to store the XML in the pact json file, so its stored in a string (along with any matchers that have been encoded as part of the test)


This is unpacked at the provider side, and reads the content type of application/xml and sends the content appropriately to the provider

Reply:
`json_encode($iContainXML)` should handle escaping etc though right?

I guess we need to know if pact needs UTF-8 escaped, but XML does not and JSON spec does not, so I'm guessing not. 😄

Reply:
I updated the builder to use the Functional Options pattern.

Reply:
- About UTF-8 encoding:
  - According to this [safety note](https://github.com/pact-foundation/pact-reference/blob/master/rust/pact_ffi/src/mock_server/handles.rs#L1344), I think yes, pact (in this case `pactffi_with_body`) need UTF-8 encoding.
  - According to this [doc](https://www.php.net/manual/en/function.json-encode.php#refsect1-function.json-encode-parameters) I think `json_encode` also need the value to be UTF-8 encoded.
- About UTF-8 encoding vs escaping: I'm not good at this. I don't think I handle these things at all in the code. I think we need to add special test cases and then update the code to handle these cases.